src/MongoDB/ReadPreference.c

```diff
@@ -217,25 +242,48 @@ static PHP_METHOD(ReadPreference, __construct)
 		bson_destroy(tags);
 	}
 
-	if (options && php_array_exists(options, "maxStalenessSeconds")) {
-		zend_long maxStalenessSeconds = php_array_fetchc_long(options, "maxStalenessSeconds");
-
-		if (maxStalenessSeconds != MONGOC_NO_MAX_STALENESS) {
-			if (maxStalenessSeconds < MONGOC_SMALLEST_MAX_STALENESS_SECONDS) {
-				phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT, "Expected maxStalenessSeconds to be >= %d, %" PHONGO_LONG_FORMAT " given", MONGOC_SMALLEST_MAX_STALENESS_SECONDS, maxStalenessSeconds);
-				return;
-			}
-			if (maxStalenessSeconds > INT32_MAX) {
-				phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT, "Expected maxStalenessSeconds to be <= %" PRId32 ", %" PHONGO_LONG_FORMAT " given", INT32_MAX, maxStalenessSeconds);
-				return;
+	if (options) {
+		if (php_array_exists(options, "maxStalenessSeconds")) {
```

Comment:
For the record, I'm fine with adding an extra `if (options && php_array_exists(options, "hedge"))` if you'd rather keep the diff small, given it's only two instances and a non-zero check is cheap. No objections if you prefer this, though.

Reply:
Reverted for a smaller diff 👍